<!doctype html><html lang=en><head><meta name=generator content="Hugo 0.136.3"><title>Armin Sobhani</title>
<meta name=description content="Personal web site of Armin Sobhani."><meta charset=UTF-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=X-UA-Compatible content="ie=edge"><link rel=stylesheet href=/application.eee28c6b9bcdfff2e871d61e6452c5d934440b55492219aa06a62a39fcf37681.css integrity="sha256-7uKMa5vN//LocdYeZFLF2TREC1VJIhmqBqYqOfzzdoE="><link rel=icon type=image/png href=/images/site/main-logo_hu7026777046544322096.png><meta property="og:title" content="Armin Sobhani's web site"><meta property="og:type" content="website"><meta property="og:description" content="Personal web site of Armin Sobhani."><meta property="og:image" content="/images/author/armin.png"><meta property="og:url" content="https://arminms.github.io"><meta name=twitter:card content="summary"><meta name=twitter:title content="Armin Sobhani"><script>theme=localStorage.getItem("theme-scheme")||localStorage.getItem("darkmode:color-scheme")||"light",theme=="system"&&(window.matchMedia&&window.matchMedia("(prefers-color-scheme: dark)").matches?theme="dark":theme="light"),document.documentElement.setAttribute("data-theme",theme)</script></head><body data-bs-spy=scroll data-bs-target=#top-navbar data-bs-offset=100><nav class="navbar navbar-expand-xl top-navbar shadow transparent-navbar homepage" id=top-navbar><div class=container><a class=navbar-brand href=/><img src=/images/site/main-logo_hu7026777046544322096.png id=logo alt=Logo>
Armin Sobhani</a>
<button class="navbar-toggler navbar-dark" id=navbar-toggler type=button data-bs-toggle=collapse data-bs-target=#top-nav-items aria-label=menu>
<i data-feather=menu></i></button><div class="collapse navbar-collapse dynamic-navbar" id=top-nav-items><ul class="nav navbar-nav ms-auto"><li class=nav-item><a class=nav-link href=#home>Home</a></li><li class=nav-item><a class=nav-link href=#about>About</a></li><li class=nav-item><a class=nav-link href=#experiences>Experiences</a></li><li class=nav-item><a class=nav-link href=#education>Education</a></li><li class=nav-item><a class=nav-link href=#projects>Projects</a></li><li class=nav-item><a class=nav-link href=#publications>Publications</a></li><li class="nav-item dropdown"><a class="nav-link dropdown-toggle" href=# id=navbarDropdown role=button data-bs-toggle=dropdown aria-haspopup=true aria-expanded=false>More</a><div class=dropdown-menu aria-labelledby=navbarDropdown><a class=dropdown-item href=#recent-posts>Recent Posts</a></div></li><div id=top-navbar-divider></div><li class=nav-item><a class=nav-link id=blog-link href=/posts>Posts</a></li><li class=nav-item><a class=nav-link href=https://staff.sharcnet.ca/asobhani>Terminal</a></li><li class=nav-item><a class=nav-link href=https://github.com/arminms/arminms.github.io>Source</a></li><li class="nav-item dropdown"><a class="nav-link dropdown-toggle" href=# id=themeSelector role=button data-bs-toggle=dropdown aria-haspopup=true aria-expanded=false><img id=navbar-theme-icon-svg class=theme-icon src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme"></a><div id=themeMenu class="dropdown-menu dropdown-menu-icons-only" aria-labelledby=themeSelector><a class="dropdown-item nav-link" href=# data-scheme=light><img class=theme-icon src=/icons/sun-svgrepo-com.svg width=20 alt="Light Theme">
</a><a class="dropdown-item nav-link" href=# data-scheme=dark><img class=theme-icon src=/icons/moon-svgrepo-com.svg width=20 alt="Dark Theme">
</a><a class="dropdown-item nav-link" href=# data-scheme=system><img class=theme-icon src=/icons/computer-svgrepo-com.svg width=20 alt="System Theme"></a></div></li></ul></div></div><img src=/images/site/main-logo_hu7026777046544322096.png class=d-none id=main-logo alt=Logo>
<img src=/images/site/main-logo_hu7026777046544322096.png class=d-none id=inverted-logo alt="Inverted Logo"></nav><div class="container-fluid home" id=home><style>#homePageBackgroundImageDivStyled{background-image:url(/images/site/background_hu5386931718381701170.jpg)}@media(min-width:500px) and (max-width:800px){#homePageBackgroundImageDivStyled{background-image:url(/images/site/background_hu10117415496670847798.jpg)}}@media(min-width:801px) and (max-width:1200px){#homePageBackgroundImageDivStyled{background-image:url(/images/site/background_hu11977726522435505659.jpg)}}@media(min-width:1201px) and (max-width:1500px){#homePageBackgroundImageDivStyled{background-image:url(/images/site/background_hu16565284723637571779.jpg)}}@media(min-width:1501px){#homePageBackgroundImageDivStyled{background-image:url(/images/site/background.jpg)}}</style><span class=on-the-fly-behavior></span><div id=homePageBackgroundImageDivStyled class="background container-fluid"></div><div class="container content text-center"><img src=/images/author/armin_hu13528733259637447528.png class="rounded-circle mx-auto d-block img-fluid" alt="Author Image"><h1 class=greeting>Hey! I am Armin 👋</h1><div class=typing-carousel><span id=ityped class=ityped></span>
<span class=ityped-cursor></span></div><ul id=typing-carousel-data><li>I am an HPC/ARC Scientific Software Developer</li><li>A Hardcore C++ Programmer</li><li>A Natural-born Debugger</li><li>I promote Generic and Functional Paradigms in programming</li><li>I am an advocate of Open-Source Software</li><li>My algorithmic mind adores Fugue and House musics</li></ul><a href=#about class=arrow-center aria-label="Read More - Armin 👋"><i class="arrow bounce fa fa-chevron-down"></i></a></div></div><div class="container-fluid section-holder d-flex bg-primary"><div class="container anchor p-lg-5 about-section" id=about><div class="row pt-sm-2 pt-md-4 align-self-center"><div class=col-sm-6><h3 class=p-1>Armin Sobhani</h3><h5 class=p-1>HPC Technical Consultant
at <a href=https://sharcnet.ca title=SHARCNET target=_blank rel=noopener>SHARCNET</a></h5><p class="p-1 text-justify">I am a seasoned <strong>HPC/ARC technical consultant</strong> based at <strong>Ontario Tech University</strong> with over 20+ years’ experience developing <strong>parallel scientific software</strong> and providing support and training to faculty, staff and students. My background is in <strong>Bioinformatics</strong> and <em>in-silico</em> <strong>Drug Design</strong>. I have substantial experience in developing software for HPC environments including but not limited to <em>shared</em> and <em>distributed-memory</em> architectures (<strong>Threads/OpenMP</strong>, <strong>MPI</strong>, <strong>PGAS</strong>), XPU ecosystems (<strong>CUDA</strong>, <strong>ROCm</strong>, <strong>oneAPI</strong>) and <em>heterogeneous computing</em> (<strong>OpenCL</strong>, <strong>OpenACC</strong>, <strong>SYCL</strong>). I love cooking, swimming, digital photography, listening to music and of course video gaming!</p><div class="text-container ms-auto"><ul class="social-link d-flex"><li><a href=mailto:arminms@gmail.com title=Email target=_blank rel=noopener><i class="fas fa-envelope"></i></a></li><li><a href=https://linkedin.com/in/arminms title=LinkedIn target=_blank rel=noopener><i class="fab fa-linkedin"></i></a></li><li><a href=https://twitter.com/arminms title=Twitter target=_blank rel=noopener><i class="fab fa-twitter"></i></a></li><li><a href=https://github.com/arminms title=GitHub target=_blank rel=noopener><i class="fab fa-github"></i></a></li><li><a href=https://stackoverflow.com/u/1681455 title=Stackoverflow target=_blank rel=noopener><i class="fab fa-stack-overflow"></i></a></li><li><a href=https://youtube.com/@armyxn title=YouTube target=_blank rel=noopener><i class="fab fa-youtube"></i></a></li><li><a href=https://www.facebook.com/armin.madadkar title=Facebook target=_blank rel=noopener><i class="fab fa-facebook"></i></a></li><li><a href=https://instagram.com/arm1ns0bhan1 title=Instagram target=_blank rel=noopener><i class="fab fa-instagram"></i></a></li></ul></div></div><div class="col-sm-6 pt-5 ps-md-4 ps-sm-3 pt-sm-0"><div class=row><div class="col-6 col-lg-4 p-2"><div class=circular-progress><span class=circular-progress-left><span class="circular-progress-bar circular-progress-percentage-40 circular-progress-percentage-50-delay" style=border-color:#99f></span>
</span><span class=circular-progress-right><span class="circular-progress-bar circular-progress-percentage-50" style=border-color:#99f></span></span><div class=circular-progress-value>Bioinformatics</div></div></div><div class="col-6 col-lg-4 p-2"><div class=circular-progress><span class=circular-progress-left><span class="circular-progress-bar circular-progress-percentage-40 circular-progress-percentage-50-delay" style=border-color:#c9f></span>
</span><span class=circular-progress-right><span class="circular-progress-bar circular-progress-percentage-50" style=border-color:#c9f></span></span><div class=circular-progress-value>in silico Drug Design</div></div></div><div class="col-6 col-lg-4 p-2"><div class=circular-progress><span class=circular-progress-left><span class="circular-progress-bar circular-progress-percentage-45 circular-progress-percentage-50-delay" style=border-color:#f9f></span>
</span><span class=circular-progress-right><span class="circular-progress-bar circular-progress-percentage-50" style=border-color:#f9f></span></span><div class=circular-progress-value>HPC / ARC</div></div></div><div class="col-6 col-lg-4 p-2"><div class=circular-progress><span class=circular-progress-left><span class="circular-progress-bar circular-progress-percentage-45 circular-progress-percentage-50-delay" style=border-color:#9cf></span>
</span><span class=circular-progress-right><span class="circular-progress-bar circular-progress-percentage-50" style=border-color:#9cf></span></span><div class=circular-progress-value>C/C++</div></div></div><div class="col-6 col-lg-4 p-2"><div class=circular-progress><span class=circular-progress-left><span class="circular-progress-bar circular-progress-percentage-35 circular-progress-percentage-50-delay" style=border-color:#f90></span>
</span><span class=circular-progress-right><span class="circular-progress-bar circular-progress-percentage-50" style=border-color:#f90></span></span><div class=circular-progress-value>CMake</div></div></div><div class="col-6 col-lg-4 p-2"><div class=circular-progress><span class=circular-progress-left><span class="circular-progress-bar circular-progress-percentage-25 circular-progress-percentage-50-delay" style=border-color:#9ff></span>
</span><span class=circular-progress-right><span class="circular-progress-bar circular-progress-percentage-50" style=border-color:#9ff></span></span><div class=circular-progress-value>Python</div></div></div><div class="col-6 col-lg-4 p-2"><div class=circular-progress><span class=circular-progress-left><span class="circular-progress-bar circular-progress-percentage-30 circular-progress-percentage-50-delay" style=border-color:#76b900></span>
</span><span class=circular-progress-right><span class="circular-progress-bar circular-progress-percentage-50" style=border-color:#76b900></span></span><div class=circular-progress-value>CUDA</div></div></div><div class="col-6 col-lg-4 p-2"><div class=circular-progress><span class=circular-progress-left><span class="circular-progress-bar circular-progress-percentage-25 circular-progress-percentage-50-delay" style=border-color:#ef0001></span>
</span><span class=circular-progress-right><span class="circular-progress-bar circular-progress-percentage-50" style=border-color:#ef0001></span></span><div class=circular-progress-value>ROCm</div></div></div><div class="col-6 col-lg-4 p-2"><div class=circular-progress><span class=circular-progress-left><span class="circular-progress-bar circular-progress-percentage-0 circular-progress-percentage-50-delay" style=border-color:#0068b5></span>
</span><span class=circular-progress-right><span class="circular-progress-bar circular-progress-percentage-50" style=border-color:#0068b5></span></span><div class=circular-progress-value>oneAPI</div></div></div></div></div></div></div></div><div class="container-fluid section-holder d-flex bg-secondary"><div class="container-fluid anchor pb-5 experiences-section"><h1 class=text-center><span id=experiences></span>Experiences</h1><div class="container timeline text-justify"><div class="row align-items-center d-flex"><div class="col-1 col-lg-2 text-center vertical-line d-inline-flex justify-content-center"><div class="circle fw-bold">1</div></div><div class="col-10 col-lg-8"><div class=experience-entry-heading><div class=company-heading><h5><a href=https://sharcnet.ca title=SHARCNET target=_blank rel=noopener>SHARCNET</a></h5><p class=text-muted>May 2015 - Present</p></div><p class=text-muted><i class="fa-solid fa-location-dot"></i> Faculty of Science, Ontario Tech University, Oshawa, Ontario, Canada</p><p>SHARCNET provides HPC/ARC facilities to a total of 19 academic partner institutions in the southern Ontario.</p></div><div class=positions><div class=company-heading><h5 class=designation>High-Performance Computing Technical Consultant</h5><p class=text-muted>May 2015 - Present</p></div><h6 class=text-heading>Responsibilities:</h6><ul class=justify-content-around><li>Member of the national Accelerator Team at the <a href=https://aliancecan.ca target=_blank rel=noopener>Digital Research Alliance of Canada (DRAC)</a></li><li>Provide broad HPC user support, tools, and training to faculty, staff and students at local (<a href=https://ontariotechu.ca target=_blank rel=noopener>Ontario Tech University</a>), regional (<a href=https://www.computeontario.ca/ target=_blank rel=noopener>Compute Ontario</a>) and national (<a href=https://aliancecan.ca target=_blank rel=noopener>Compute Canada/DRAC</a>) levels.</li><li>Actively participate in training and outreach programs including but not limited to New User and Refresher Seminars (NURSE), General Interest Seminars (GIS), Summer Schools on HPC, Intro to ARC, etc.</li><li>Provide programming and consulting support to build expertise in the use of <a href=https://sharcnet.ca target=_blank rel=noopener>SHARCNET</a>/<a href=https://alliancecan.ca target=_blank rel=noopener>DRAC</a> facilities.</li><li>Contribute to <a href=https://www.sharcnet.ca/my/research/programming target=_blank rel=noopener>SHARCNET&rsquo;s Dedicated Programming</a> projects.</li><li><strong>Primary Technologies Used:</strong> Ansible, Apptainer/Singularity, BLAS, Boost, Catch2, CCDB, CMake, CUDA, cuBLAS, cuSOLVER, Docker, DirectX, FlexiBLAS, FPGA, gcc/g++/gdb, Git, GitHub Actions, GNU GSL, HoloLens, Hugo, IBM LSF, LAPACK, MPI, Ninja, Nix, NVIDIA HPC SDK, oneAPI, OpenGL, OpenMP, OTRS, PGAS, Python, ROCm, Slack, SLURM, Sphinx, STL, SYCL, TBB, Thrust, vcpkg, VSCode, Vulkan</li></ul></div></div></div><div class="row horizontal-line"><div class="col-1 col-lg-2 timeline-side-div"><div class=corner></div></div><div class="col-10 col-lg-8"><hr></div><div class="col-1 col-lg-2 timeline-side-div"><div class=corner></div></div></div><div class="row align-items-center justify-content-end d-flex"><div class="col-10 col-lg-8"><div class=experience-entry-heading><div class=company-heading><h5><a href=https://www.ocadu.ca/ title="OCAD University" target=_blank rel=noopener>OCAD University</a></h5><p class=text-muted>March 2015 - April 2015</p></div><p class=text-muted><i class="fa-solid fa-location-dot"></i> Toronto, Ontario, Canada</p><p>OCAD U is the largest and most comprehensive art, design and media university in Canada.</p></div><div class=positions><div class=company-heading><h5 class=designation>Computer Support Assistant</h5><p class=text-muted>March 2015 - April 2015</p></div><h6 class=text-heading>Responsibilities:</h6><ul class=justify-content-around><li>Provide technical support and assist in the training of faculties, staff, and students regarding the use of computers.</li><li>Research and install updates on current hardware and software.</li><li>Maintain currency in relevant technologies, including operating systems, software, computer hardware, networks, mobile and assistive devices.</li></ul></div></div><div class="col-1 col-lg-2 text-center vertical-line d-inline-flex justify-content-center"><div class="circle fw-bold">2</div></div></div><div class="row horizontal-line"><div class="col-1 col-lg-2 timeline-side-div"><div class=corner></div></div><div class="col-10 col-lg-8"><hr></div><div class="col-1 col-lg-2 timeline-side-div"><div class=corner></div></div></div><div class="row align-items-center d-flex"><div class="col-1 col-lg-2 text-center vertical-line d-inline-flex justify-content-center"><div class="circle fw-bold">3</div></div><div class="col-10 col-lg-8"><div class=experience-entry-heading><div class=company-heading><h5><a href=https://bsc.es title="Barcelona Supercomputing Center (BSC)" target=_blank rel=noopener>Barcelona Supercomputing Center (BSC)</a></h5><p class=text-muted>2010 - 2014</p></div><p class=text-muted><i class="fa-solid fa-location-dot"></i> Barcelona, Catalonia, Spain</p><p>BSC is Spain&rsquo;s leading supercomputing centre and specialises in High Performance Computing.</p></div><div class=positions><div class=company-heading><h5 class=designation>Senior Researcher</h5><p class=text-muted>2010 - 2014</p></div><h6 class=text-heading>Responsibilities:</h6><ul class=justify-content-around><li>Conduct research projects and write grant proposals for the national supercomputing facility of Spain.</li><li>Collaborate on PELE project, a large parallel Monte Carlo sampling molecular simulation codebase supported by the European Research Council (ERC) starter grant (€1,500,000), as part of a variable team size of 5-10 people.</li><li>Conduct interviews and brainstorming sessions with the team.</li><li>Deliver formal presentations and tutorials.</li><li>Provide consultation in purchasing new hardware.</li><li><strong>Primary Technologies Used:</strong> Object-Oriented Programming, Functional Programming, Generic Programming, Template Metaprogramming (TMP), Design Patterns, Intel® Fortran Compiler, gcc/g++, STL, Boost, Python, MPI, OpenMP, CUDA, Intel® MKL and IPP, blaze, Wt, JavaScript, HTML5, CSS, Bootstrap, MySQL, MFC, OpenGL, MTK++, Gnuplot, Boost.Test, Eclipse CDT, Code::Blocks, SVN, Doxygen, Bitnami stack, Redmine, VirtualBox, OpenSUSE, SLURM, IBM LSF, modules</li></ul></div></div></div><div class="row horizontal-line"><div class="col-1 col-lg-2 timeline-side-div"><div class=corner></div></div><div class="col-10 col-lg-8"><hr></div><div class="col-1 col-lg-2 timeline-side-div"><div class=corner></div></div></div><div class="row align-items-center justify-content-end d-flex"><div class="col-10 col-lg-8"><div class=experience-entry-heading><div class=company-heading><h5><a href=https://www.ut.ac.ir/en title="University of Tehran" target=_blank rel=noopener>University of Tehran</a></h5><p class=text-muted>2007 - 2010</p></div><p class=text-muted><i class="fa-solid fa-location-dot"></i> Department of Bioinformatics, Institute of Biochemistry & Biophysics (IBB), Tehran, Iran</p><p>The University of Tehran is the oldest and most prominent Iranian university.</p></div><div class=positions><div class=company-heading><h5 class=designation>Assistant Professor</h5><p class=text-muted>2007 - 2010</p></div><h6 class=text-heading>Responsibilities:</h6><ul class=justify-content-around><li>Carry out research projects and publish in the field of bioinformatics.</li><li>Teach graduate students (B.Sc. and Ph.D. students of bioinformatics, biophysics, biochemistry and biotechnology).</li><li>Supervise M.Sc. theses and Ph.D. dissertations.</li><li>Develope various programs in the field of bioinformatics.</li><li><strong>Primary Technologies Used:</strong> OOP, Design Patterns, Visual Studio/C++, MFC, OpenGL, SVN, Doxygen, FreeImage, Microsoft Compiled HTML Help, InstallShield</li></ul></div></div><div class="col-1 col-lg-2 text-center vertical-line d-inline-flex justify-content-center"><div class="circle fw-bold">4</div></div></div><div class="row horizontal-line"><div class="col-1 col-lg-2 timeline-side-div"><div class=corner></div></div><div class="col-10 col-lg-8"><hr></div><div class="col-1 col-lg-2 timeline-side-div"><div class=corner></div></div></div><div class="row align-items-center d-flex"><div class="col-1 col-lg-2 text-center vertical-line d-inline-flex justify-content-center"><div class="circle fw-bold">5</div></div><div class="col-10 col-lg-8"><div class=experience-entry-heading><div class=company-heading><h5><a href=https://en.iums.ac.ir/ title="Iran University of Medical Sciences" target=_blank rel=noopener>Iran University of Medical Sciences</a></h5><p class=text-muted>2001 - 2007</p></div><p class=text-muted><i class="fa-solid fa-location-dot"></i> Department of Pharmacology, School of Medicine, Tehran, Iran</p><p>Iran University of Medical Sciences is a high ranked medical university in Iran.</p></div><div class=positions><div class=company-heading><h5 class=designation>Assistant Professor</h5><p class=text-muted>2001 - 2007</p></div><h6 class=text-heading>Responsibilities:</h6><ul class=justify-content-around><li>Carry out research projects and publish in the field of <em>in silico</em> drug design.</li><li>Teach undergraduate students (nursing, midwiferies, paramedics, …).</li><li>Teach graduate students (MD, Ph.D. candidates in pharmacology, biochemistry, physiology, etc.).</li><li>Supervise M.Sc. theses and Ph.D. dissertations.</li><li>Develope software controlling data acquisition (DAQ) hardware for monitoring animal behavior.</li><li>Administer university&rsquo;s mail, web, Firewall/Proxy, DNS, DHCP and TACACS servers.</li><li>Manage and maintain network/internet infrastructure of the university.</li><li>Administer and maintain Sun SPARC, HP and IBM servers.</li><li>Provide consultation in purchasing hardware and software.</li><li><strong>Primary Technologies Used:</strong> Apache, Cisco ASA, DNS, IIS, LAMPP, Microsoft Exchange, MS ISA Server, Squid, SunOS, Sun Solaris, TACACS, Zimbra.</li></ul></div></div></div><div class="row horizontal-line"><div class="col-1 col-lg-2 timeline-side-div"><div class=corner></div></div><div class="col-10 col-lg-8"><hr></div><div class="col-1 col-lg-2 timeline-side-div"><div class=corner></div></div></div><div class="row align-items-center justify-content-end d-flex"><div class="col-10 col-lg-8"><div class=experience-entry-heading><div class=company-heading><h5><a href=http://www.nigeb.ac.ir/web/en title="National Institute of Genetic Engineering & Biotechnology (NIGEB)" target=_blank rel=noopener>National Institute of Genetic Engineering & Biotechnology (NIGEB)</a></h5><p class=text-muted>1994 - 2001</p></div><p class=text-muted><i class="fa-solid fa-location-dot"></i> Tehran, Iran</p><p>Top Iranian research institute in the field of genetic engineering and biotechnology.</p></div><div class=positions><div class=company-heading><h5 class=designation>System and Network Administrator</h5><p class=text-muted>1994 - 2001</p></div><h6 class=text-heading>Responsibilities:</h6><ul class=justify-content-around><li>Administer institute&rsquo;s mail (sendmail), gopher, web (Apache), Telnet, FTP, DNS, NIS servers.</li><li>Administer and manage bioinformatics software and databases (IntelliGenetics IGSuite, Entrez, Swiss-Prot, PDB, etc.).</li><li>Administer and maintain Sun SPARC servers.</li><li>Manage and maintain network/internet infrastructure of the institute.</li><li>Perform hardware and software troubleshooting.</li><li>Perform regular (incremental) backups.</li><li>Deliver lectures, tutorials, workshops and practical classes.</li><li><strong>Primary Technologies Used:</strong> Apache, DNS, FTP, gopher, NIS, sendmail, SunOS, Sun Solaris.</li></ul></div></div><div class="col-1 col-lg-2 text-center vertical-line d-inline-flex justify-content-center"><div class="circle fw-bold">6</div></div></div></div></div></div><div class="container-fluid section-holder d-flex bg-primary"><div class="container-fluid anchor pb-5 education-section"><h1 class=text-center><span id=education></span>Education</h1><div class=container><table class=education-info-table><tbody><tr><td class=icon><div class=hline></div><div class=icon-holder><i class="fas fa-university"></i></div></td><td class=line><div></div></td><td class=details><div class="degree-info card"><div class=row><div class="col-lg-10 col-md-8"><h5><a href="https://www.upc.edu/en?set_language=en" title="Universitat Politecnica De Catalunya" target=_blank rel=noopener>Universitat Politecnica De Catalunya</a></h5></div><div class="timeframe col-lg-2 col-md-4">June 2011</div></div><h6>Programming and tUning Massively Parallel Systems, PUMPS Summer School</h6></div></td></tr><tr><td class=icon><div class=hline></div><div class=icon-holder><i class="fas fa-microscope"></i></div></td><td class=line><div></div></td><td class=details><div class="degree-info card"><div class=row><div class="col-lg-10 col-md-8"><h5><a href=https://en.iums.ac.ir/ title="Iran University of Medical Sciences" target=_blank rel=noopener>Iran University of Medical Sciences</a></h5></div><div class="timeframe col-lg-2 col-md-4">1995-2001</div></div><h6>Ph.D. in Pharmacology</h6><h6><span class=text-heading>CGPA: </span><span>3.53</span> out of <span>4</span></h6><div class=publications><h6 class=text-heading>Publications:</h6><ul><li><a href=http://www.ualberta.ca/~csps/JPPS5%281%29/M.Mahmoudian/peganum.pdf title="An in vitro evaluation of human DNA topoisomerase I inhibition by Peganum harmala L. seeds extract and its β-carboline alkaloids." target=_blank rel=noopener>An in vitro evaluation of human DNA topoisomerase I inhibition by Peganum harmala L. seeds extract and its β-carboline alkaloids.</a></li><li>Cytotoxicity of Peganum harmala L. seeds extract and its relationship with content of β-carboline alkaloids.</li></ul></div><div class=taken-courses><h6 class=text-heading>Taken Courses:</h6><table><thead><th class=course-name-header>Course Name</th><th>Total Credit</th><th>Obtained Credit</th></thead><tbody><tr class=course><td>Biostatistics</td><td>4</td><td>4</td></tr><tr class=course><td>Computer</td><td>4</td><td>4</td></tr><tr class=course><td>General Pharmacology</td><td>4</td><td>3.4</td></tr><tr class="course hidden-course"><td>Advanced Toxicology</td><td>4</td><td>3.8</td></tr></tbody></table><button type=button class="btn btn-link show-more-btn pt-0 ms-1" id=show-more-btn>Show More</button>
<button type=button class="btn btn-link show-more-btn hidden pt-0 ms-1" id=show-less-btn>Show Less</button></div></div></td></tr><tr><td class=icon><div class=hline></div><div class=icon-holder><i class="fas fa-graduation-cap"></i></div></td><td class=line><div></div></td><td class=details><div class="degree-info card"><div class=row><div class="col-lg-10 col-md-8"><h5><a href=https://en.tums.ac.ir/en title="Tehran University of Medical Sciences" target=_blank rel=noopener>Tehran University of Medical Sciences</a></h5></div><div class="timeframe col-lg-2 col-md-4">1988-1995</div></div><h6>Pharm.D.</h6><h6><span class=text-heading>CGPA: </span><span>3.32</span> out of <span>4</span></h6><div class=taken-courses><h6 class=text-heading>Taken Courses:</h6><table><thead><th class=course-name-header>Course Name</th><th>Total Credit</th><th>Obtained Credit</th></thead><tbody><tr class=course><td>Basic Mathematics and Introduction to Statistics</td><td>4</td><td>3.5</td></tr><tr class=course><td>Biostatistics</td><td>4</td><td>3.1</td></tr><tr class=course><td>Pharmacology</td><td>4</td><td>3.6</td></tr><tr class="course hidden-course"><td>Biochemistry</td><td>4</td><td>3.8</td></tr><tr class="course hidden-course"><td>Biophysics</td><td>4</td><td>3.4</td></tr></tbody></table><button type=button class="btn btn-link show-more-btn pt-0 ms-1" id=show-more-btn>Show More</button>
<button type=button class="btn btn-link show-more-btn hidden pt-0 ms-1" id=show-less-btn>Show Less</button></div></div></td></tr><tr><td class=icon><div class=hline></div><div class=icon-holder><i class="fas fa-school"></i></div></td><td class=line><div></div></td><td class=details><div class="degree-info card"><div class=row><div class="col-lg-10 col-md-8"><h5>Alborz High School</h5></div><div class="timeframe col-lg-2 col-md-4">1983-1987</div></div><h6>Secondary School Certificate</h6><h6><span class=text-heading>GPA: </span><span>4.61</span> out of <span>5</span></h6></div></td></tr></tbody></table></div></div></div><div class="container-fluid section-holder d-flex bg-secondary"><div class="container-fluid anchor pb-5 projects-section" id=projects><h1 class=text-center><span id=projects></span>Projects</h1><div class="container ms-auto text-center"><div class="btn-group flex-wrap" role=group id=project-filter-buttons><button type=button class="btn btn-dark project-filtr-control" data-filter=all>
All
</button>
<button type=button class="btn btn-dark project-filtr-control" data-filter=professional>
Professional
</button>
<button type=button class="btn btn-dark project-filtr-control" data-filter=academic>
Academic
</button>
<button type=button class="btn btn-dark project-filtr-control" data-filter=hobby>
Hobby</button></div></div><div class="container filtr-projects"><div class=row id=project-card-holder><div class="col-sm-12 col-md-6 col-lg-4 p-2 filtr-item" data-category='all, Wasm,WebAssembly,SDL2,Game of Life'><div class="card mt-1"><div class=card><a class=card-header href=https://github.com/arminms/gol2p target=_blank rel=noopener><div><div class=d-flex><h5 class="card-title mb-0">gol2p</h5></div><div class=sub-title><span>Owner</span>
<span>July 2024 - Present</span></div></div></a><div class="card-body text-justify pt-1 pb-1"><p>A C++ implementation of Conway&rsquo;s Game of Life using SDL2 with support for WebAssembly (Wasm) via Emscripten.</p><div class=project-card-footer><div class=project-tags-holder><span class="badge btn-info">Wasm
</span><span class="badge btn-info">WebAssembly
</span><span class="badge btn-info">SDL2
</span><span class="badge btn-info">Game of Life</span></div><div class=project-btn-holder><a class="github-button project-btn d-none" href=https://github.com/arminms/gol2p data-icon=octicon-standard data-show-count=true aria-label="Star gol2p">Star</a></div></div></div></div></div></div><div class="col-sm-12 col-md-6 col-lg-4 p-2 filtr-item" data-category='all, academic,HPC,PRNG,random'><div class="card mt-1"><div class=card><a class=card-header href=https://github.com/arminms/p2rng target=_blank rel=noopener><div><div class=d-flex><h5 class="card-title mb-0">p2rng</h5></div><div class=sub-title><span>Owner</span>
<span>October 2023 - Present</span></div></div></a><div class="card-body text-justify pt-1 pb-1"><p>A modern header-only C++ library for parallel algorithmic (pseudo) random number generation supporting OpenMP, CUDA, ROCm and oneAPI.</p><div class=project-card-footer><div class=project-tags-holder><span class="badge btn-info">academic
</span><span class="badge btn-info">HPC
</span><span class="badge btn-info">PRNG
</span><span class="badge btn-info">random</span></div><div class=project-btn-holder><a class="github-button project-btn d-none" href=https://github.com/arminms/p2rng data-icon=octicon-standard data-show-count=true aria-label="Star p2rng">Star</a></div></div></div></div></div></div><div class="col-sm-12 col-md-6 col-lg-4 p-2 filtr-item" data-category='all, academic,HPC,XPU,data-parallel'><div class="card mt-1"><div class=card><a class=card-header href=https://github.com/arminms/one4all target=_blank rel=noopener><div><div class=d-flex><h5 class="card-title mb-0">one4all</h5></div><div class=sub-title><span>Owner</span>
<span>Mar 2023 - Present</span></div></div></a><div class="card-body text-justify pt-1 pb-1"><p>A framework to streamline developing for CUDA, ROCm, oneAPI and Parallel STL at the same time.</p><div class=project-card-footer><div class=project-tags-holder><span class="badge btn-info">academic
</span><span class="badge btn-info">HPC
</span><span class="badge btn-info">XPU
</span><span class="badge btn-info">data-parallel</span></div><div class=project-btn-holder><a class="github-button project-btn d-none" href=https://github.com/arminms/one4all data-icon=octicon-standard data-show-count=true aria-label="Star one4all">Star</a></div></div></div></div></div></div><div class="col-sm-12 col-md-6 col-lg-4 p-2 filtr-item" data-category='all, academic,plotting,data-science'><div class="card mt-1"><div class=card><a class=card-header href=https://github.com/arminms/g3p target=_blank rel=noopener><div><div class=d-flex><img class=card-img-xs src=/images/sections/projects/g3p_logo.svg alt=g3p><h5 class="card-title mb-0">g3p</h5></div><div class=sub-title><span>Owner</span>
<span>Mar 2023 - Present</span></div></div></a><div class="card-body text-justify pt-1 pb-1"><p>A modern header-only C++ interface library for gnuplot. It is the most natural and intuitive way of adding gnuplot support into any C++ program.</p><div class=project-card-footer><div class=project-tags-holder><span class="badge btn-info">academic
</span><span class="badge btn-info">plotting
</span><span class="badge btn-info">data-science</span></div><div class=project-btn-holder><a class="github-button project-btn d-none" href=https://github.com/arminms/g3p data-icon=octicon-standard data-show-count=true aria-label="Star g3p">Star</a></div></div></div></div></div></div><div class="col-sm-12 col-md-6 col-lg-4 p-2 filtr-item" data-category='all, academic,bioinformatics,NGS'><div class="card mt-1"><div class=card><a class=card-header href=https://github.com/arminms/amstools target=_blank rel=noopener><div><div class=d-flex><h5 class="card-title mb-0">amstools</h5></div><div class=sub-title><span>Owner</span>
<span>May 2022 - Present</span></div></div></a><div class="card-body text-justify pt-1 pb-1"><p>Collection of high-performance biological sequence tools developed in C/C++.</p><div class=project-card-footer><div class=project-tags-holder><span class="badge btn-info">academic
</span><span class="badge btn-info">bioinformatics
</span><span class="badge btn-info">NGS</span></div><div class=project-btn-holder><a class="github-button project-btn d-none" href=https://github.com/arminms/amstools data-icon=octicon-standard data-show-count=true aria-label="Star amstools">Star</a></div></div></div></div></div></div><div class="col-sm-12 col-md-6 col-lg-4 p-2 filtr-item" data-category='all, academic,professional,HPC'><div class="card mt-1"><div class=card><a class=card-header href=https://github.com/sharcnet/vscode-hpc target=_blank rel=noopener><div><div class=d-flex><h5 class="card-title mb-0">vscode-hpc</h5></div><div class=sub-title><span>Owner</span>
<span>Jan 2022 - Sep 2022</span></div></div></a><div class="card-body text-justify pt-1 pb-1"><p>Sample code for remote development on HPC clusters with VSCode.</p><div class=project-card-footer><div class=project-tags-holder><span class="badge btn-info">academic
</span><span class="badge btn-info">professional
</span><span class="badge btn-info">HPC</span></div><div class=project-btn-holder><a class="github-button project-btn d-none" href=https://github.com/sharcnet/vscode-hpc data-icon=octicon-standard data-show-count=true aria-label="Star vscode-hpc">Star</a></div></div></div></div></div></div><div class="col-sm-12 col-md-6 col-lg-4 p-2 filtr-item" data-category='all, academic,optimization,RBF,RSM'><div class="card mt-1"><div class=card><a class=card-header href=https://github.com/sharcnet/rsmtk target=_blank rel=noopener><div><div class=d-flex><h5 class="card-title mb-0">RSMTK</h5></div><div class=sub-title><span>Owner</span>
<span>Mar 2019 - Feb 2022</span></div></div></a><div class="card-body text-justify pt-1 pb-1"><p>A modern C++ library and toolkit for global and local optimization using radial basis function response surface models.</p><div class=project-card-footer><div class=project-tags-holder><span class="badge btn-info">academic
</span><span class="badge btn-info">optimization
</span><span class="badge btn-info">RBF
</span><span class="badge btn-info">RSM</span></div><div class=project-btn-holder><a class="github-button project-btn d-none" href=https://github.com/sharcnet/rsmtk data-icon=octicon-standard data-show-count=true aria-label="Star RSMTK">Star</a></div></div></div></div></div></div><div class="col-sm-12 col-md-6 col-lg-4 p-2 filtr-item" data-category='all, academic,NetCDF,HDF5'><div class="card mt-1"><div class=card><a class=card-header href=https://git.sharcnet.ca/DP/spins-tools target=_blank rel=noopener><div><div class=d-flex><h5 class="card-title mb-0">spins-tools</h5></div><div class=sub-title><span>Owner</span>
<span>Aug 2017 - Jun 2018</span></div></div></a><div class="card-body text-justify pt-1 pb-1"><p>A set of tools for converting SPINS&rsquo; output format to self-describing data formats (e.g. NetCDF and HDF5).</p><div class=project-card-footer><div class=project-tags-holder><span class="badge btn-info">academic
</span><span class="badge btn-info">NetCDF
</span><span class="badge btn-info">HDF5</span></div><div class=project-btn-holder><a class="github-button project-btn d-none" href=https://git.sharcnet.ca/DP/spins-tools data-icon=octicon-standard data-show-count=true aria-label="Star spins-tools">Star</a></div></div></div></div></div></div><div class="col-sm-12 col-md-6 col-lg-4 p-2 filtr-item" data-category='all, hobby,HoloLens,molecule,3D'><div class="card mt-1"><div class=card><a class=card-header href=https://github.com/arminms/HoliMoli target=_blank rel=noopener><div><div class=d-flex><img class=card-img-xs src=/images/sections/projects/holimoli_hu424635093497067621.png alt=HoliMoli><h5 class="card-title mb-0">HoliMoli</h5></div><div class=sub-title><span>Owner</span>
<span>May 2016 - Dec 2016</span></div></div></a><div class="card-body text-justify pt-1 pb-1"><p>The molecular hologram viewer for Microsoft HoloLens.</p><div class=project-card-footer><div class=project-tags-holder><span class="badge btn-info">hobby
</span><span class="badge btn-info">HoloLens
</span><span class="badge btn-info">molecule
</span><span class="badge btn-info">3D</span></div><div class=project-btn-holder><a class="github-button project-btn d-none" href=https://github.com/arminms/HoliMoli data-icon=octicon-standard data-show-count=true aria-label="Star HoliMoli">Star</a></div></div></div></div></div></div><div class="col-sm-12 col-md-6 col-lg-4 p-2 filtr-item" data-category='all, academic,professional'><div class="card mt-1"><div class=card><a class=card-header href=https://github.com/arminms/hashkat-v2 target=_blank rel=noopener><div><div class=d-flex><img class=card-img-xs src=/images/sections/projects/hashkat.svg alt="Hashkat Version 2 (#k@-v2)"><h5 class="card-title mb-0">Hashkat Version 2 (#k@-v2)</h5></div><div class=sub-title><span>Developer</span>
<span>Aug 2015 - Mar 2016</span></div></div></a><div class="card-body text-justify pt-1 pb-1"><p>HPC version of original <a href=http://hashkat.org/ target=_blank rel=noopener>#k@</a> (Kinetic Analysis of Twitter-like Network).</p><div class=project-card-footer><div class=project-tags-holder><span class="badge btn-info">academic
</span><span class="badge btn-info">professional</span></div><div class=project-btn-holder><a class="github-button project-btn d-none" href=https://github.com/arminms/hashkat-v2 data-icon=octicon-standard data-show-count=true aria-label="Star Hashkat Version 2 (#k@-v2)">Star</a></div></div></div></div></div></div><div class="col-sm-12 col-md-6 col-lg-4 p-2 filtr-item" data-category='all, academic,hobby,molecule,3D'><div class="card mt-1"><div class=card><a class=card-header href=https://github.com/arminms/maral target=_blank rel=noopener><div><div class=d-flex><h5 class="card-title mb-0">Maral</h5></div><div class=sub-title><span>Owner</span>
<span>May 2012 - Mar 2015</span></div></div></a><div class="card-body text-justify pt-1 pb-1"><p>Maral is a header-only template based C++11 molecular API.</p><div class=project-card-footer><div class=project-tags-holder><span class="badge btn-info">academic
</span><span class="badge btn-info">hobby
</span><span class="badge btn-info">molecule
</span><span class="badge btn-info">3D</span></div><div class=project-btn-holder><a class="github-button project-btn d-none" href=https://github.com/arminms/maral data-icon=octicon-standard data-show-count=true aria-label="Star Maral">Star</a></div></div></div></div></div></div><div class="col-sm-12 col-md-6 col-lg-4 p-2 filtr-item" data-category='all, academic,professional,drug design'><div class="card mt-1"><div class=card><a class=card-header href=https://pele.bsc.es target=_blank rel=noopener><div><div class=d-flex><h5 class="card-title mb-0">PELE Web Server</h5></div><div class=sub-title><span>Developer</span>
<span>May 2011 - Feb 2014</span></div></div></a><div class="card-body text-justify pt-1 pb-1"><p>Web interface for Protein Energy Landscape Exploration program written in C++.</p><div class=project-card-footer><div class=project-tags-holder><span class="badge btn-info">academic
</span><span class="badge btn-info">professional
</span><span class="badge btn-info">drug design</span></div><div class=project-btn-holder><span><a class="btn btn-outline-info btn-sm" href=https://pele.bsc.es target=#>Details</a></span></div></div></div></div></div></div></div></div></div></div><div class="container-fluid section-holder d-flex bg-primary"><div class="container-fluid anchor pb-5 publications-section" id=publications><h1 class=text-center><span id=publications></span>Publications</h1><div class="container ms-auto text-center"><div class="btn-group flex-wrap" role=pub-group id=publication-filter-buttons><button type=button class="btn btn-dark pub-filtr-control" data-filter=pub-all>
All
</button>
<button type=button class="btn btn-dark pub-filtr-control" data-filter=pub-selected>
Selected
</button>
<button type=button class="btn btn-dark pub-filtr-control" data-filter=pub-bio>
Bioinformatics
</button>
<button type=button class="btn btn-dark pub-filtr-control" data-filter=pub-comp>
Computational
</button>
<button type=button class="btn btn-dark pub-filtr-control" data-filter=pub-cadd>
Drug Design</button></div></div><div class="container filtr-publications"><div class=row id=publication-card-holder><div class="col-12 p-2 pub-filtr-item" data-category=pub-all,pub-cadd,pub-comp><div class="card mt-3"><div class=card-header><h5 class="card-title mb-0">Insights into the human A1 adenosine receptor from molecular dynamics simulation: structural study in the presence of lipid membrane</h5><div class=sub-title><span><a href=https://www.springer.com/journal/44>Medicinal Chemistry Research</a></span>
<span class=ms-2>19 July 2015</span></div><div class=authors><span class=me-2><a href>Mahboubeh Mansourian</a></span>
<span class=me-2><a href>Karim Mahnam</a></span>
<span class=me-2><a href>Armin Madadkar-Sobhani</a></span>
<span class=me-2><a href>Afshin Fassihi</a></span>
<span class=me-2><a href>Lotfollah Saghaie</a></span></div></div><div class=card-body><p>Homology modeling, molecular docking and molecular dynamics (MD) simulation methods were used to build a reliable model for A1AR (as one of the G protein-coupled receptors—GPCRs) and to explore the structural features and binding mechanism of ligands to this receptor. A model of A1AR was built and inserted in a hydrated lipid bilayer, and 20-ns MD simulation was performed to examine the stability of the best model. In this study, RG-14718 as the best A1AR agonist and bamifylline as a selective antagonist of A1AR have been docked into the active site of the A1AR. After docking, two 20-ns MD simulation was performed on the A1AR–ligand complex to explore effects of the presence of lipid membrane in the vicinity of the A1AR–ligand complex. At the end of the MD simulation, a change in the position and orientation of the ligand in the binding site was observed. This important observation indicated that the application of MD simulation after docking of ligands is useful. Thr270, His278 and Asn70 were crucial residues for hydrogen bonds with these ligands. Phe171, Glu172, Tyr271 and Ile274 were determined to be involved in ligand–receptor binding. The results obtained are in good agreement with most of the site-directed mutagenesis data reported by others. Our results show that molecular modeling and rational drug design for adenosine targets is a possible approach.</p></div><div class=card-footer><div class=tags><span class="btn badge btn-info ms-1 p-2">MD
</span><span class="btn badge btn-info ms-1 p-2">GPCR
</span><span class="btn badge btn-info ms-1 p-2">adenosine receptor</span></div><div class=details-btn><a class="btn btn-outline-info ms-1 ps-2 mb-2" href=http://dx.doi.org/10.1007/s00044-015-1409-6 target=_blank rel=noopener role=button>Details</a></div></div></div></div><div class="col-12 p-2 pub-filtr-item" data-category=pub-all,pub-selected,pub-bio,pub-comp><div class="card mt-3"><div class=card-header><h5 class="card-title mb-0">SDRL: a sequence-dependent protein side-chain rotamer library</h5><div class=sub-title><span><a href="https://pubs.rsc.org/en/journals/journalissues/mb#!issueid=mb013012&amp;type=archive&amp;issnprint=1742-206x">Molecular BioSystems</a></span>
<span class=ms-2>20 April 2015</span></div><div class=authors><span class=me-2><a href>Mohammad Taghizadeh</a></span>
<span class=me-2><a href>Bahram Goliae</a></span>
<span class=me-2><a href>Armin Madadkar-Sobhani</a></span></div></div><div class=card-body><p>Since the introduction of the first protein side-chain rotamer library (RL) almost half a century ago, RLs have been components of many programs and algorithms in structural bioinformatics. Based on the dependence of side-chain dihedral angles on the local backbone, three types of RLs have been identified: backbone-independent, secondary-structure-dependent and backbone-dependent. In all previous studies, the effect of sequence specificity on side-chain conformational preferences was neglected. In the effort to develop a new class of RLs, we considered that the side-chain conformation of the central residue in each triplet on a protein backbone depends on the sequence of the triplet; therefore, we developed a sequence-dependent rotamer library (SDRL). To accomplish this, 400 possible triplet sequences for 18 natural amino acids as the central residue, which corresponds to 7200 triplet sequences in total, were considered. Searching the set of 11 546 selected PDB entries for the 7200 triplet sequences resulted in 2 364 541 instances occurring for 18 amino acids. Our results show that Leu and Val experience minimal impact from the adjacent residues in adopting side-chain conformations. Cys, Ile, Trp, His, Asp, Met, Glu, Gln, Arg and Lys, on the other hand, adopt their side-chain conformations mostly based on the adjacent residues on the backbone. The remaining residue types were moderately dependent on the adjacent residues. Using the new library, side-chain repacking algorithms can find preferred conformations of each residue more easily than with other backbone-independent RLs.</p></div><div class=card-footer><div class=tags><span class="btn badge btn-info ms-1 p-2">protein
</span><span class="btn badge btn-info ms-1 p-2">rotamer-library
</span><span class="btn badge btn-info ms-1 p-2">side-chain</span></div><div class=details-btn><a class="btn btn-outline-info ms-1 ps-2 mb-2" href=http://dx.doi.org/10.1039/c5mb00057b target=_blank rel=noopener role=button>Details</a></div></div></div></div><div class="col-12 p-2 pub-filtr-item" data-category=pub-all,pub-cadd,pub-comp><div class="card mt-3"><div class=card-header><h5 class="card-title mb-0">QSAR and docking analysis of A2B adenosine receptor antagonists based on non-xanthine scaffold</h5><div class=sub-title><span><a href=https://www.springer.com/journal/44>Medicinal Chemistry Research</a></span>
<span class=ms-2>15 July 2014</span></div><div class=authors><span class=me-2><a href>Mahboubeh Mansourian</a></span>
<span class=me-2><a href>Afshin Fassihi</a></span>
<span class=me-2><a href>Lotfollah Saghaie</a></span>
<span class=me-2><a href>Armin Madadkar-Sobhani</a></span>
<span class=me-2><a href>Karim Mahnam</a></span>
<span class=me-2><a href>Maryam Abbasi</a></span></div></div><div class=card-body><p>Partial least squares (PLS) method as ligand-based method was applied for building quantitative structure–activity relationships (QSAR) regression model to predict the inhibitory activity of some A2B antagonists. The accuracy and predictability of the developed model were evaluated by several validation methods using external and internal test sets and also the criteria recommended by Tropsha and Roy were met. The result of the PLS model had a high statistical quality (R 2 = 0.936 and Q 2 = 0.867) for predicting the activity of the compounds. Evaluation of a test set of seven compounds with the developed PLS model revealed that this model is reliable and has a good predictability. Because of high correlation between the predicted and experimental values of activity, PLS model proved to be a highly predictive QSAR approach. Also, the reliability of the model was assessed through docking for the selected antagonists as structure-based method. A potential binding site of A2BAR was verified according to the previous studies of site-directed mutagenesis. Phe173, Glu174, His251, Asn254, Lys269, Ile276, and His 280 were determined to be involved in ligand–receptor interactions.</p></div><div class=card-footer><div class=tags><span class="btn badge btn-info ms-1 p-2">adenosine receptor
</span><span class="btn badge btn-info ms-1 p-2">PLS
</span><span class="btn badge btn-info ms-1 p-2">QSAR</span></div><div class=details-btn><a class="btn btn-outline-info ms-1 ps-2 mb-2" href=http://dx.doi.org/10.1007/s00044-014-1133-7 target=_blank rel=noopener role=button>Details</a></div></div></div></div><div class="col-12 p-2 pub-filtr-item" data-category=pub-all,pub-all,pub-cadd,pub-comp><div class="card mt-3"><div class=card-header><h5 class="card-title mb-0">A structural model of the anaphase promoting complex co-activator (Cdh1) and in silico design of inhibitory compounds</h5><div class=sub-title><span><a href=http://rps.mui.ac.ir/index.php/jrps>Research in Pharmaceutical Sciences</a></span>
<span class=ms-2>February 2015</span></div><div class=authors><span class=me-2><a href>H. Rahimi</a></span>
<span class=me-2><a href>B. Negahdari</a></span>
<span class=me-2><a href>M.A. Shokrgozar</a></span>
<span class=me-2><a href>R. Mahdian</a></span>
<span class=me-2><a href>A. Foroumadi</a></span>
<span class=me-2><a href>M. Kafshdouzi Amin</a></span>
<span class=me-2><a href>M. Karimipoor</a></span></div></div><div class=card-body><p>Anaphase promoting complex (APC) controls cell cycle and chromosome segregation. The APC activation occurs after binding of co-activators, cdh1 and cdc20. Cdh1 plays a role in cancer pathogenesis and is known as a potential drug target. The main aim of this study was prediction of 3D structure of cdh1 and designing the inhibitory compounds based on the structural model. First, 3D structure of cdh1 was predicted by means of homology modelling and molecular dynamics tools, MODELLER and Gromacs package, respectively. Then, inhibitory compounds were designed using virtual screening and molecular docking by means AutoDock package. The overall structure of cdh1 is propeller like and each DW40 repeat contains four antiparallel beta-sheets. Moreover, binding pocket of the inhibitory compounds was determined. The results might be helpful in finding a suitable cdh1 inhibitor for the treatment of cancer.</p></div><div class=card-footer><div class=tags><span class="btn badge btn-info ms-1 p-2">APC
</span><span class="btn badge btn-info ms-1 p-2">AutoDock
</span><span class="btn badge btn-info ms-1 p-2">MODELLER</span></div><div class=details-btn><a class="btn btn-outline-info ms-1 ps-2 mb-2" href=http://rps.mui.ac.ir/index.php/jrps/article/download/1605/1563 target=_blank rel=noopener role=button>Details</a></div></div></div></div><div class="col-12 p-2 pub-filtr-item" data-category=pub-all,pub-bio,pub-comp><div class="card mt-3"><div class=card-header><h5 class="card-title mb-0">Automatic classification of highly related Malate Dehydrogenase and L-Lactate Dehydrogenase based on 3D-pattern of active sites</h5><div class=sub-title><span><a href=https://pbiosci.ut.ac.ir/>Progress in Biological Sciences</a></span>
<span class=ms-2>December 2014</span></div><div class=authors><span class=me-2><a href>Amir Rahimi</a></span>
<span class=me-2><a href>Armin Madadkar-Sobhani</a></span>
<span class=me-2><a href>Rouzbeh Touserkani</a></span>
<span class=me-2><a href>Bahram Goliaei</a></span></div></div><div class=card-body><p>Accurate protein function prediction is an important subject in bioinformatics, especially where sequentially and structurally similar proteins have different functions. Malate dehydrogenase and L-lactate dehydrogenase are two evolutionary related enzymes, which exist in a wide variety of organisms. These enzymes are sequentially and structurally similar and share common active site residues, spatial patterns and molecular mechanisms. Here, we study various features of the active site cavity of 229 PDB chain entries and try to classify them automatically by various classifiers including the support vector machine, k nearest neighbour and random forest methods. The results show that the support vector machine yields the highest predictive performance among mentioned classifiers. Despite very close and conserved patterns among Malate dehydrogenases and L-lactate dehydrogenases, the SVM predicts the function efficiently and achieves 0.973 Matthew’s correlation coefficient and 0.987 F-score. The same approach can be used in other enzyme families for automatic discrimination between homologous enzymes with common active site elements, however, acting on different substrates.</p></div><div class=card-footer><div class=tags><span class="btn badge btn-info ms-1 p-2">3D-motifs
</span><span class="btn badge btn-info ms-1 p-2">SVM</span></div><div class=details-btn><a class="btn btn-outline-info ms-1 ps-2 mb-2" href=http://dx.doi.org/10.22059/PBS.2014.52303 target=_blank rel=noopener role=button>Details</a></div></div></div></div><div class="col-12 p-2 pub-filtr-item" data-category=pub-all,pub-cadd,pub-comp,pub-selected><div class="card mt-3"><div class=card-header><h5 class="card-title mb-0">PELE web server: atomistic study of biomolecular systems at your fingertips</h5><div class=sub-title><span><a href=https://academic.oup.com/nar>Nucleic Acids Research</a></span>
<span class=ms-2>1 July 2013</span></div><div class=authors><span class=me-2><a href>Armin Madadkar-Sobhani</a></span>
<span class=me-2><a href>Victor Guallar</a></span></div></div><div class=card-body><p>PELE, Protein Energy Landscape Exploration, our novel technology based on protein structure prediction algorithms and a Monte Carlo sampling, is capable of modelling the all-atom protein–ligand dynamical interactions in an efficient and fast manner, with two orders of magnitude reduced computational cost when compared with traditional molecular dynamics techniques. PELE’s heuristic approach generates trial moves based on protein and ligand perturbations followed by side chain sampling and global/local minimization. The collection of accepted steps forms a stochastic trajectory. Furthermore, several processors may be run in parallel towards a collective goal or defining several independent trajectories; the whole procedure has been parallelized using the Message Passing Interface. Here, we introduce the PELE web server, designed to make the whole process of running simulations easier and more practical by minimizing input file demand, providing user-friendly interface and producing abstract outputs (e.g. interactive graphs and tables). The web server has been implemented in C++ using Wt (<a href=http://www.webtoolkit.eu target=_blank rel=noopener>http://www.webtoolkit.eu</a>) and MySQL (<a href=http://www.mysql.com target=_blank rel=noopener>http://www.mysql.com</a>). The PELE web server, accessible at <a href=http://pele.bsc.es target=_blank rel=noopener>http://pele.bsc.es</a>, is free and open to all users with no login requirement.</p></div><div class=card-footer><div class=tags><span class="btn badge btn-info ms-1 p-2">C++
</span><span class="btn badge btn-info ms-1 p-2">MC
</span><span class="btn badge btn-info ms-1 p-2">MySQL
</span><span class="btn badge btn-info ms-1 p-2">Web server
</span><span class="btn badge btn-info ms-1 p-2">Wt</span></div><div class=details-btn><a class="btn btn-outline-info ms-1 ps-2 mb-2" href=http://dx.doi.org/10.1093/nar/gkt454 target=_blank rel=noopener role=button>Details</a></div></div></div></div><div class="col-12 p-2 pub-filtr-item" data-category=pub-all,pub-cadd><div class="card mt-3"><div class=card-header><h5 class="card-title mb-0">Design, synthesis and biological evaluation of novel anti-cytokine 1,2,4-triazine derivatives</h5><div class=sub-title><span><a href=https://www.sciencedirect.com/journal/bioorganic-and-medicinal-chemistry>Bioorganic and Medicinal Chemistry</a></span>
<span class=ms-2>1 July 2013</span></div><div class=authors><span class=me-2><a href>Mehdi Khoshneviszadeh</a></span>
<span class=me-2><a href>Mohammad H. Ghahremani</a></span>
<span class=me-2><a href>Alireza Foroumadi</a></span>
<span class=me-2><a href>Ramin Miri</a></span>
<span class=me-2><a href>Omidreza Firuzi</a></span>
<span class=me-2><a href>Armin Madadkar-Sobhani</a></span>
<span class=me-2><a href>Najmeh Edrak</a></span>
<span class=me-2><a href>Maliheh Parsa</a></span>
<span class=me-2><a href>Abbas Shafiee</a></span></div></div><div class=card-body><p>A series of 16 novel 1,2,4-triazine derivatives bearing hydrazone moiety (7a–7p) have been designed, synthesized and evaluated for their activity to inhibit IL-1β and TNF-α production. All compounds are reported for the first time. The chemical structures of all compounds were confirmed by spectroscopic methods and elemental analyzes. Most of the synthesized compounds were proved to have potent anti-cytokine activity and low toxicity on PBMC and MCF-7 cell lines. Compounds 7f, 7k, 7l and 7j presented simultaneously good levels of inhibition of both cytokines. Moreover, compound 7l exhibited good anti-inflammatory effect in carrageenan-induced rat paw edema. The results of Western blotting demonstrated that the anti-cytokine potential of compound 7l is mainly mediated through the inhibition of p38 MAPK signaling pathway. Molecular docking was performed to position compound 7l into p38α binding site in order to explore the potential target. The information of this work might be helpful for the design and synthesis of novel scaffold toward the development of new therapeutic agent to fight against inflammatory diseases.</p></div><div class=card-footer><div class=tags><span class="btn badge btn-info ms-1 p-2">IL-1β
</span><span class="btn badge btn-info ms-1 p-2">synthesis
</span><span class="btn badge btn-info ms-1 p-2">TNF-α</span></div><div class=details-btn><a class="btn btn-outline-info ms-1 ps-2 mb-2" href=http://dx.doi.org/10.1016/j.bmc.2013.08.009 target=_blank rel=noopener role=button>Details</a></div></div></div></div><div class="col-12 p-2 pub-filtr-item" data-category=pub-all,pub-bio,pub-comp,pub-selected><div class="card mt-3"><div class=card-header><h5 class="card-title mb-0">Efficacy of function specific 3D-motifs in enzyme classification according to their EC-numbers</h5><div class=sub-title><span><a href=https://www.sciencedirect.com/journal/journal-of-theoretical-biology>Journal of Theoretical Biology</a></span>
<span class=ms-2>7 November 2013</span></div><div class=authors><span class=me-2><a href>Amir Rahimi</a></span>
<span class=me-2><a href>Armin Madadkar-Sobhani</a></span>
<span class=me-2><a href>Rouzbeh Touserkani</a></span>
<span class=me-2><a href>Bahram Goliaei</a></span></div></div><div class=card-body><p>Due to the increasing number of protein structures with unknown function originated from structural genomics projects, protein function prediction has become an important subject in bioinformatics. Among diverse function prediction methods, exploring known 3D-motifs, which are associated with functional elements in unknown protein structures is one of the most biologically meaningful methods. Homologous enzymes inherit such motifs in their active sites from common ancestors. However, slight differences in the properties of these motifs, results in variation in the reactions and substrates of the enzymes. In this study, we examined the possibility of discriminating highly related active site patterns according to their EC-numbers by 3D-motifs. For each EC-number, the spatial arrangement of an active site, which has minimum average distance to other active sites with the same function, was selected as a representative 3D-motif. In order to characterize the motifs, various points in active site elements were tested. The results demonstrated the possibility of predicting full EC-number of enzymes by 3D-motifs. However, the discriminating power of 3D-motifs varies among different enzyme families and depends on selecting the appropriate points and features.</p></div><div class=card-footer><div class=tags><span class="btn badge btn-info ms-1 p-2">3D-motifs
</span><span class="btn badge btn-info ms-1 p-2">classification
</span><span class="btn badge btn-info ms-1 p-2">enzyme</span></div><div class=details-btn><a class="btn btn-outline-info ms-1 ps-2 mb-2" href=http://dx.doi.org/10.1016/j.jtbi.2013.07.003 target=_blank rel=noopener role=button>Details</a></div></div></div></div><div class="col-12 p-2 pub-filtr-item" data-category=pub-all,pub-cadd><div class="card mt-3"><div class=card-header><h5 class="card-title mb-0">Phenylimino-2H-chromen-3-carboxamide derivatives as novel small molecule inhibitors of β-secretase (BACE1)</h5><div class=sub-title><span><a href=https://www.sciencedirect.com/journal/bioorganic-and-medicinal-chemistry>Bioorganic and Medicinal Chemistry</a></span>
<span class=ms-2>15 April 2013</span></div><div class=authors><span class=me-2><a href>Najmeh Edraki</a></span>
<span class=me-2><a href>Omidreza Firuzi</a></span>
<span class=me-2><a href>Alireza Foroumadi</a></span>
<span class=me-2><a href>Ramin Miri</a></span>
<span class=me-2><a href>Armin Madadkar-Sobhani</a></span>
<span class=me-2><a href>Mehdi Khoshneviszadeh</a></span>
<span class=me-2><a href>Abbas Shafiee</a></span></div></div><div class=card-body><p>The inhibition of β secretase (BACE1) is potentially important approach to treatment of Alzheimer disease (AD). A novel series of 4-bromophenyl piperazine derivatives coupled to the phenylimino-2H-chromen-3-carboxamide scaffold were investigated as BACE1 inhibitors in this study. Docking study suggested that the phenyl-imino group of the scaffold establishes favorable π–π stacking interaction with side chain of Phe108 of flap pocket. Some of the docking proposed derivatives were synthesized and evaluated for BACE1 inhibitory activity using a FRET-based assay. High BACE1 inhibitory activities were observed from derivatives containing fused heteroaromtic groups attached through the aliphatic linkage to the N4-piperazine moiety, which may be attributed to the engagement of effective interactions with S1–S′1 sub-pocket residues. Of the most potent compounds, 9e displayed an IC50 value for BACE1 of 98 nM. Some of these derivatives demonstrated good inhibitory activity on Aβ production in N2a-APPswe cells at 5 and 10 μM. These compounds might be considered as promising BACE1 inhibitory agents that could lower Aβ production in AD.</p></div><div class=card-footer><div class=tags><span class="btn badge btn-info ms-1 p-2">Alzheimer
</span><span class="btn badge btn-info ms-1 p-2">BACE1
</span><span class="btn badge btn-info ms-1 p-2">FRET</span></div><div class=details-btn><a class="btn btn-outline-info ms-1 ps-2 mb-2" href=http://dx.doi.org/10.1016/j.bmc.2013.01.064 target=_blank rel=noopener role=button>Details</a></div></div></div></div><div class="col-12 p-2 pub-filtr-item" data-category=pub-all,pub-comp,pub-selected,pub-cadd><div class="card mt-3"><div class=card-header><h5 class="card-title mb-0">Linear and nonlinear QSAR modeling of 1,3,8-substituted-9-deazaxanthines as potential selective A2BAR antagonists</h5><div class=sub-title><span><a href=https://www.tandfonline.com/journals/ienz20>Medicinal Chemistry Research</a></span>
<span class=ms-2>13 January 2013</span></div><div class=authors><span class=me-2><a href>Mahboubeh Mansourian</a></span>
<span class=me-2><a href>Lotfollah Saghaie</a></span>
<span class=me-2><a href>Afshin Fassihi</a></span>
<span class=me-2><a href>Armin Madadkar-Sobhani</a></span>
<span class=me-2><a href>Karim Mahnam</a></span></div></div><div class=card-body><p>A QSAR study of 74 derivatives of 1,3,8-substituted-9-deazaxanthines as potent and selective A2B adenosine receptor (A2BAR) antagonists is described. pKi of all the studied compounds were acquired by three linear and nonlinear methods namely stepwise multiple linear regression, partial least squares (PLS), and general regression neural networks (GRNN). The performances of developed models were tested by several external and internal validation methods and also the criteria recommended by Tropsha and Roy. Predictability and possible overfitting in the resulting models were examined by cross-validation. Results revealed the significant role of topological and geometrical descriptors in binding of the studied compounds to A2BAR. PLS and GRNN models had good statistical qualities with PLS showing better performance (R 2 = 0.863 and Q 2 = 0.817). Applicability domain of the models was also defined. The prediction results were in good agreement with the experimental data.</p></div><div class=card-footer><div class=tags><span class="btn badge btn-info ms-1 p-2">A2BAR
</span><span class="btn badge btn-info ms-1 p-2">GRNN
</span><span class="btn badge btn-info ms-1 p-2">PLS
</span><span class="btn badge btn-info ms-1 p-2">QSAR</span></div><div class=details-btn><a class="btn btn-outline-info ms-1 ps-2 mb-2" href=http://dx.doi.org/10.1007/s00044-012-0453-8 target=_blank rel=noopener role=button>Details</a></div></div></div></div><div class="col-12 p-2 pub-filtr-item" data-category=pub-all,pub-comp,pub-selected,pub-cadd><div class="card mt-3"><div class=card-header><h5 class="card-title mb-0">Computational evaluation of some indenopyrazole derivatives as anticancer compounds; application of QSAR and docking methodologies</h5><div class=sub-title><span><a href=https://www.springer.com/journal/44>Journal of Enzyme Inhibition and Medicinal Chemistry</a></span>
<span class=ms-2>14 Oct 2011</span></div><div class=authors><span class=me-2><a href>Mohsen Shahlaei</a></span>
<span class=me-2><a href>Afshin Fassihi</a></span>
<span class=me-2><a href>Lotfollah Saghaie</a></span>
<span class=me-2><a href>Elham Arkan</a></span>
<span class=me-2><a href>Armin Madadkar-Sobhani</a></span>
<span class=me-2><a href>Alireza Pourhossein</a></span></div></div><div class=card-body><p>A computational procedure was performed on some indenopyrazole derivatives. Two important procedures in computational drug discovery, namely docking for modeling ligand-receptor interactions and quantitative structure activity relationships were employed. MIA-QSAR analysis of the studied derivatives produced a model with high predictability. The developed model was then used to evaluate the bioactivity of 54 proposed indenopyrazole derivatives. In order to confirm the obtained results through this ligand-based method, docking was performed on the selected compounds. An ADME–Tox evaluation was also carried out to search for more suitable compounds. Satisfactory bioactivities and ADME–Tox proﬁles for two of the compounds, namely 62 and S13, propose that further studies should be performed on such devoted chemical structures.</p></div><div class=card-footer><div class=tags><span class="btn badge btn-info ms-1 p-2">ADME–Tox
</span><span class="btn badge btn-info ms-1 p-2">docking
</span><span class="btn badge btn-info ms-1 p-2">QSAR</span></div><div class=details-btn><a class="btn btn-outline-info ms-1 ps-2 mb-2" href=http://dx.doi.org/10.3109/14756366.2011.618991 target=_blank rel=noopener role=button>Details</a></div></div></div></div></div></div></div></div><div class="container-fluid section-holder d-flex bg-secondary"><div class="container-fluid anchor pb-5 recent-posts-section"><h1 class=text-center><span id=recent-posts></span>Recent Posts</h1><div class=container><div class=row id=recent-post-cards><div class=post-card><div class=card><div class=card-head><a href=/posts/videos/gol2p/ class=post-card-link><img class=card-img-top src=/posts/videos/gol2p/images/PXL_20240901_113740870.jpg alt="Hero Image"></a></div><div class=card-body><a href=/posts/videos/gol2p/ class=post-card-link><h5 class=card-title>The Emergence of WebAssembly (Wasm) in Scientific Computing</h5><p class="card-text post-summary">I presented this webinar on August 7th, 2024, as a part of a series of weekly Compute Ontario Colloquia. Developed collaboratively by major browser vendors, including Mozilla, Google, Microsoft, and Apple, WebAssembly (Wasm) addresses the limitations of traditional web programming languages like JavaScript. But what makes it so compelling for scientists? First, Wasm allows code written in languages like C/C++, Fortran or Rust to be compiled into its instruction format and run directly in the browser, making it accessible to anyone without installation hassles and eliminating the need for external servers. Second, with Wasm, developers can recycle existing code with near-native performance but without the hassle of rewriting it in JavaScript. Join us as we explore how Wasm is reshaping scientific workflows and empowering researchers worldwide.</p></a></div><div class=card-footer><span class=float-start>Friday, September 20, 2024
</span><a href=/posts/videos/gol2p/ class="float-end btn btn-outline-info btn-sm">Read</a></div></div></div><div class=post-card><div class=card><div class=card-head><a href=/posts/videos/p2rng/ class=post-card-link><img class=card-img-top src=/posts/videos/p2rng/images/PXL_20220718_233206312.jpg alt="Hero Image"></a></div><div class=card-body><a href=/posts/videos/p2rng/ class=post-card-link><h5 class=card-title>p2rng – A C++ Parallel Random Number Generator Library for the Masses</h5><p class="card-text post-summary">I presented this webinar on October 18th, 2023, as a part of a series of weekly Compute Ontario Colloquia. p2rng (https://github.com/arminms/p2rng) is a modern header-only C++ library for parallel algorithmic (pseudo) random number generation supporting OpenMP, CUDA, ROCm and oneAPI. Playing fair, mostly required for debugging and unit testing, is one of the unique features of p2rng. That means using the same seed and distribution you always get the same sequence of random numbers on all supported platforms. p2rng provides parallel versions of STL’s std::generate() and std::generate_n() algorithms with the same interface. In this seminar we first start with a quick review of preliminary concepts about algorithmic random number generators in general and parallelization techniques in particular. Then we continue with the standard way of generating random numbers with STL algorithms and how we can turn them into parallel version using p2rng.</p></a></div><div class=card-footer><span class=float-start>Monday, October 23, 2023
</span><a href=/posts/videos/p2rng/ class="float-end btn btn-outline-info btn-sm">Read</a></div></div></div><div class=post-card><div class=card><div class=card-head><a href=/posts/videos/one4all/ class=post-card-link><img class=card-img-top src=/posts/videos/one4all/images/p1030224.jpg alt="Hero Image"></a></div><div class=card-body><a href=/posts/videos/one4all/ class=post-card-link><h5 class=card-title>CUDA, ROCm, oneAPI – All for One or One for All?</h5><p class="card-text post-summary">I presented this webinar on April 19th, 2023, as a part of a series of weekly Compute Ontario Colloquia. For a long time, CUDA was the platform of choice for developing applications running on NVIDIA’s GPUs. That is starting to change in recent years with the introduction of AMD’s ROCm and Intel’s oneAPI which both support GPUs by other vendors. While ROCm targets both AMD and NVIDIA GPUs, using the recently released drivers by CodePlay, oneAPI applications can run on NVIDIA and AMD in addition to Intel’s GPUs. The question this seminar is trying to answer is if in 2023 you want to start a project targeting GPUs, what would be your platform of choice? Should you go with one or all of them? Later in the seminar, a boilerplate framework named one4all will be introduced that streamlines the process of developing applications targeting all the above platforms. Unit testing with Catch2 as well as benchmarking with Google benchmark are already supported by the framework.</p></a></div><div class=card-footer><span class=float-start>Friday, April 21, 2023
</span><a href=/posts/videos/one4all/ class="float-end btn btn-outline-info btn-sm">Read</a></div></div></div></div></div></div></div><footer id=footer class="container-fluid text-center align-content-center footer pb-2"><div class="container pt-5"><div class="row text-start"><div class="col-md-4 col-sm-12"><h5>Navigation</h5><ul><li class=nav-item><a class=smooth-scroll href=https://armin.sobhani.me/#about>About</a></li><li class=nav-item><a class=smooth-scroll href=https://armin.sobhani.me/#experiences>Experiences</a></li><li class=nav-item><a class=smooth-scroll href=https://armin.sobhani.me/#education>Education</a></li><li class=nav-item><a class=smooth-scroll href=https://armin.sobhani.me/#projects>Projects</a></li><li class=nav-item><a class=smooth-scroll href=https://armin.sobhani.me/#publications>Publications</a></li><li class=nav-item><a class=smooth-scroll href=https://armin.sobhani.me/#recent-posts>Recent Posts</a></li><li class=nav-item><a class=smooth-scroll href=https://staff.sharcnet.ca/asobhani>Terminal</a></li><li class=nav-item><a class=smooth-scroll href=https://github.com/arminms/arminms.github.io>Source</a></li></ul></div><div class="col-md-4 col-sm-12"><h5>Contact me:</h5><ul><li><a href=mailto:arminms@gmail.com target=_blank rel=noopener><span><i class="fas fa-envelope"></i></span> <span>arminms@gmail.com</span></a></li><li><a href=https://github.com/arminms target=_blank rel=noopener><span><i class="fab fa-github"></i></span> <span>arminms</span></a></li><li><a href=https://www.linkedin.com/in/arminms target=_blank rel=noopener><span><i class="fab fa-linkedin"></i></span> <span>Armin Sobhani</span></a></li></ul></div></div></div></footer><script src=/application.33f83dfcec37774b96b9fe56fc0b8c5a1fe16a1181df936ecf9b47c34bc3b960.js integrity="sha256-M/g9/Ow3d0uWuf5W/AuMWh/hahGB35Nuz5tHw0vDuWA=" defer></script><script data-name=BMC-Widget data-cfasync=false src=https://cdnjs.buymeacoffee.com/1.0.0/widget.prod.min.js data-id=armyxn data-description="Support me on Buy Me a Coffee!" data-message="Buy me a coffee!" data-color=#FFDD00 data-position=Right data-x_margin=10 data-y_margin=18></script></body></html>